derive_builder_core/src/build_method.rs

```diff
@@ -65,6 +68,11 @@ impl<'a> ToTokens for BuildMethod<'a> {
             BuilderPattern::Immutable => quote!(&self),
         };
         let doc_comment = &self.doc_comment;
+        let default = if let Some(ref expr) = self.struct_default { 
+            quote!(let __default : #target_ty = #expr;)
```

Comment:
I think I found a way to generate the code like this:
```rust
Foo {
    foo: 42,
    ..#struct_default // <-- only emitted if struct_default is set.
}
```

I will commit this as soon as I checked a few other things. :-)